### DISCUSSION TITLE
Scale render window is not locked when rendering

### DISCUSSION THREAD

**Discussion by PeterJared** [lllyasviel/Fooocus#3083] [Ideas]
Currently, we can resize the render window, which is very handy when doing widescreen renders as it allows us to save space from the default square render window. However, the aspect ratio is not locked. This means that as soon as the render starts, it returns to a square aspect ratio. Once the render finishes, it reverts to our customized aspect ratio, and this cycle continues. In batch renders, this means the render window is continuously swapping the aspect ratio, moving around the UI below. For example, if we want to edit the image prompt settings, they will move up and down continuously, which is more noticeable when batch rendering is quick.
I would suggest locking the aspect ratio once it is customized by users.

**Comment by mashb1t** [comment]
![screencapture-localhost-7865-2024-06-02-21_08_13](https://github.com/lllyasviel/Fooocus/assets/9307310/e1238cac-c295-4373-a646-d5aa7533715d)

On generation start, the final gallery will be hidden and instead the progress_window + progress_gallery are shown, which are not resizable. All three elements have a fixed height of 768.
=> Those are not the same elements.

![image](https://github.com/lllyasviel/Fooocus/assets/9307310/f37463e7-84be-4e53-aeb1-d85fe140ec60)
This doesn't really work/look good. Your feedback how to optimize the UI is welcome.

**Comment by PeterJared** [comment]
I guess you mean the generation window is not resizable to match the normal display, meaning we are stuck with the square aspect ratio. It's not a huge problem if it's difficult or not possible to solve.

I just noticed the main problem is that the generate forever option returns to the normal display screen by default after each two images, even if it is set to infinite. This is what creates the flickering effect when the renders are fast. Now that I understand what is going on, I can set a specific number of images I want.

But for a more user-friendly setup when setup to render forever:
- Lock the aspect ratio in the render windows (it seems not possible, but it's not a big problem).
- When the user sets it to generate forever, invert the image order, displaying the first one instead of the last one (this would be a good default in general to see the last renders), and only clear the screen after 32 images instead of after 2.

---

**Reply by mashb1t** [reply]
Thanks for your opinion. 1 is not possible for the container (i mean vae preview image aspect ratio is already locked) and 2 can't be done in isolation for all generated image cycles as each generation batch (depenting on image number), even for generate forever, is isolated and the screen has to be cleared after each cycle, as it'd only JavaScript on the button and the return event automatically clicks the button for you, nothing more.
It's not easy to optimize the UI without upgrading Gradio, and this is a big task, which i currently sadly don't have the capacity for.

**Comment by PeterJared** [comment]
I think now that is clear the behaviour is better to leave it as it is. I notice this has more implications that I thought.
By default is set to 2 images, if used generate forever and we modify the prompt or settings, it only will read the changes each 2 images. If we set it to more, for example 6, it will change only any update after 6 images, that is not practical etc. Because the limitations you commented, In this case, its better simply leave the default squared aspect ratio to have more flexibility in more important issues.
